### PR TITLE
Add some types to views

### PR DESCRIPTION
**Pull request by adamchainz**
# I have made things!

Found some view functions in `ModelAdmin` that take `request: Any` and decided to expand the search for all such view functions. Then added types to all of them.

## Related issues

n/a

### CODE REVIEW

File: django-stubs/contrib/gis/admin/options.pyi [hunk 1]
```diff
@@ -36,7 +37,7 @@ class GeoModelAdmin(ModelAdmin):
     widget: Any = ...
     @property
     def media(self) -> Any: ...
-    def formfield_for_dbfield(self, db_field: Any, request: Any, **kwargs: Any) -> Any: ...
+    def formfield_for_dbfield(self, db_field: Any, request: HttpRequest, **kwargs: Any) -> Any: ...
```
Comment (adamchainz) [inline django-stubs/contrib/gis/admin/options.pyi]: we already had this type in the superclass: https://github.com/typeddjango/django-stubs/blob/44806740da9f71de325c1e50586ab6b21fd2c318/django-stubs/contrib/admin/options.pyi#L117

File: django-stubs/contrib/gis/sitemaps/views.pyi [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
+    request: HttpRequest,
+    label: str,
+    model: str,
+    field_name: Optional[str] = ...,
+    compress: bool = ...,
+    using: str = ...,
+) -> HttpResponse: ...
+def kmz(
+    request: HttpRequest, label: str, model: str, field_name: Optional[str] = ..., using: str = ...
+) -> HttpResponse: ...
```
Comment (adamchainz) [inline django-stubs/contrib/gis/sitemaps/views.pyi:15]: These two views are poorly documented in Django itself: https://docs.djangoproject.com/en/4.1/ref/contrib/gis/sitemaps/

I looked in the source to find types: https://github.com/django/django/blob/2bc47d7fe95ec9fc62bc5c0d2408944e5269d9d0/django/contrib/gis/sitemaps/views.py#L10

And at the related URL patterns, which seem only used in Django’s test suite: https://github.com/django/django/blob/2bc47d7fe95ec9fc62bc5c0d2408944e5269d9d0/tests/gis_tests/geoapp/urls.py#L17-L28

File: django-stubs/contrib/gis/views.pyi [hunk 3]
```diff
@@ -1,3 +1,6 @@
-from typing import Any, Optional
+from typing import Dict, Optional, Type
 
-def feed(request: Any, url: Any, feed_dict: Optional[Any] = ...) -> Any: ...
+from django.contrib.gis.feeds import Feed
+from django.http import HttpRequest, HttpResponse
+
+def feed(request: HttpRequest, url: str, feed_dict: Optional[Dict[str, Type[Feed]]] = ...) -> HttpResponse: ...
```
Comment (adamchainz) [inline django-stubs/contrib/gis/views.pyi:6]: Also poorly documented: https://docs.djangoproject.com/en/4.1/ref/contrib/gis/feeds/

Again poked around to determine types: https://github.com/django/django/blob/84206607d6bfd61e7f7a88b51163ffd4153e3b5a/django/contrib/gis/views.py#L23

File: django-stubs/views/debug.pyi [hunk 4]
```diff
@@ -14,7 +14,11 @@ class CallableSettingWrapper:
     def __init__(self, callable_setting: Union[Callable, Type[Any]]) -> None: ...
 
 def technical_500_response(
-    request: Any, exc_type: Any, exc_value: Any, tb: Any, status_code: int = ...
+    request: HttpRequest,
+    exc_type: Optional[Type[BaseException]],
+    exc_value: Optional[BaseException],
+    tb: Optional[TracebackType],
+    status_code: int = ...,
 ) -> HttpResponse: ...
```
Comment (adamchainz) [inline django-stubs/views/debug.pyi:22]: The args are all passed on to `ExceptionReporter` which we have types for below

https://github.com/django/django/blob/84206607d6bfd61e7f7a88b51163ffd4153e3b5a/django/views/debug.py#L59-L72

File: django-stubs/contrib/admin/options.pyi [hunk 5]
```diff
@@ -190,16 +190,16 @@ class ModelAdmin(BaseModelAdmin[_ModelT]):
     def media(self) -> Media: ...
     def get_model_perms(self, request: HttpRequest) -> Dict[str, bool]: ...
     def get_form(
-        self, request: Any, obj: Optional[_ModelT] = ..., change: bool = ..., **kwargs: Any
+        self, request: HttpRequest, obj: Optional[_ModelT] = ..., change: bool = ..., **kwargs: Any
```
Comment (adamchainz) [inline django-stubs/contrib/admin/options.pyi:193]: For all the Django admin changes I read the source, `request` seems consistently used for `HttpRequest` throughout, as you'd hope

File: django-stubs/contrib/gis/admin/options.pyi (hunk 1 above)
Comment (sobolevn) [reply django-stubs/contrib/gis/admin/options.pyi]: Maybe we can remove it from the subtype then?

Comment (adamchainz) [reply django-stubs/contrib/gis/admin/options.pyi]: You're right, removed.

**Review by sobolevn — APPROVED** [review]
I've never used this part of Django! Thank you.

**Comment by adamchainz** [comment]
Me neither, I came for the admin, but searched for `request: Any` and fixed the other places. There are certainly some dark corners inside `django.contrib.gis`.

**Pull request merged**